gui/packages/desktop/src/main/index.js

```diff
@@ -20,19 +21,38 @@ import {
   defaultSettings,
   defaultTunnelStateTransition,
 } from './daemon-rpc';
-import type { RelayList, TunnelState, TunnelStateTransition, Settings } from './daemon-rpc';
+import type {
+  AppVersionInfo,
+  RelayList,
+  Settings,
+  TunnelState,
+  TunnelStateTransition,
+} from './daemon-rpc';
 
 import ReconnectionBackoff from './reconnection-backoff';
 import { resolveBin } from './proc';
 
 const RELAY_LIST_UPDATE_INTERVAL = 60 * 60 * 1000;
+const VERSION_UPDATE_INTERVAL = 24 * 60 * 60 * 1000;
 
 const DAEMON_RPC_PATH =
   process.platform === 'win32' ? '//./pipe/Mullvad VPN' : '/var/run/mullvad-vpn';
 
 type AppQuitStage = 'unready' | 'initiated' | 'ready';
 
+export type CurrentAppVersionInfo = {
+  gui: string,
+  daemon: string,
+  isConsistent: boolean,
+};
+
+export type AppUpgradeInfo = {
```

Comment:
FYI I don't know how to call that structure otherwise, so I made it `AppUpgradeInfo`...